src/python/pants/goal/context.py

```diff
@@ -155,6 +155,20 @@ def __str__(self):
     ident = Target.identify(self.targets())
     return 'Context(id:{}, targets:{})'.format(ident, self.targets())
 
+  def set_target_root_count_in_runtracker(self):
+    """Sets the target root count in the run tracker's daemon stats object."""
+    # N.B. `self._target_roots` is always an expanded list of `Target` objects as
+    # provided by `GoalRunner`.
+    target_count = len(self._target_roots)
+    self.run_tracker.pantsd_stats.set_target_root_size(target_count)
+    return target_count
+
+  def set_affected_target_count_in_runtracker(self):
```

Comment:
Rather than exposing these as public methods on `Context`, is there somewhere else we could put them that would not potentially give the impression that they are intended for Task developers to call? We should be aiming to shrink the Context API if possible.

Reply:
I'd assumed making them non-`:API: public` would suffice for that - but I've made them private and wrapped that with a contextmanager to tighten this up a bit. let me know what you think.